### PR TITLE
Support PI in core deployer

### PR DESCRIPTION
**Pull request by asaj**
### Description

_What's included in this PR?_

### Drive-by changes

_Are there any minor or drive-by changes also included?_

### Related issues

- Fixes #[issue number here]

### Backward compatibility

_Are these changes backward compatible?_

Yes
No

_Are there any infrastructure implications, e.g. changes that would prohibit deploying older commits using this infra tooling?_

None
Yes

### Testing

_What kind of testing have these changes undergone?_

None
Manual
Unit Tests

### CODE REVIEW

File: typescript/sdk/src/providers/MultiProvider.ts [hunk 1]
```diff
@@ -39,12 +39,23 @@ export class MultiProvider {
...
+    }
+    this.logger = debug(options?.loggerName || 'hyperlane:MultiProvider');
+  }
+
+  addChain(metadata: ChainMetadata): void {
+    if (this.getKnownChainNames().includes(metadata.name)) {
+      throw new Error(`Duplicate chain name: ${name}`);
+    }
+    this.metadata[metadata.name] = metadata;
+    // TODO: This is O(n^2)
```
Comment (jmrossy) [inline typescript/sdk/src/providers/MultiProvider.ts:53]: Yeah but when n < 100 it's irrelevant right? The mem requirements of keeping a map make it a wash.

File: typescript/sdk/src/providers/MultiProvider.ts [hunk 2]
```diff
@@ -39,12 +39,23 @@ export class MultiProvider {
...
     options: MultiProviderOptions = {},
   ) {
-    this.metadata = chainMetadata;
+    for (const chain of Object.values(chainMetadata)) {
+      this.addChain(chain);
+    }
+    this.logger = debug(options?.loggerName || 'hyperlane:MultiProvider');
+  }
+
+  addChain(metadata: ChainMetadata): void {
```
Comment (jmrossy) [inline typescript/sdk/src/providers/MultiProvider.ts:48]: You'll need to check if useSharedSigner is true and if so add an entry in `this.signers` for the new chain. I realize the whole shared signer thing is awkward but it was a requirement in some places.

File: typescript/sdk/src/core/HyperlaneCore.ts [hunk 3]
```diff
@@ -135,21 +143,24 @@ export class HyperlaneCore extends HyperlaneApp<CoreContracts> {
...
-          return mailbox.parseLog(log);
-        } catch (e) {
-          return undefined;
-        }
-      })
-      .filter(
-        (log): log is ethers.utils.LogDescription =>
-          !!log && log.name === 'Dispatch',
-      );
+    const filterLogs = (eventName: string) => {
```
Comment (jmrossy) [inline typescript/sdk/src/core/HyperlaneCore.ts:146]: Did you maybe plan to use this function more than once, but then realize you didn't need to? Wondering why change this code at all?

File: typescript/sdk/src/providers/MultiProvider.ts (hunk 1 above)
Comment (asaj) [reply typescript/sdk/src/providers/MultiProvider.ts:53]: Yeah for sure I don't think this matters, just wanted to call it out

**Pull request closed without merging**